src/us/kbase/workspace/kbase/ArgUtils.java

```diff
@@ -113,6 +113,7 @@ public static Provenance processProvenance(final WorkspaceUser user,
 			return p;
 		}
 		for (final ProvenanceAction a: actions) {
+			// TODO PROV what if a is null?
```

Comment:
There's a number of these `TODO PROV`s around. Normally I'd take care of them in the same PR, but this PR is pretty huge already, so I'm putting them off until later PRs.